src/backend/base/langflow/components/vectorstores/OpenSearch.py

```diff
@@ -0,0 +1,260 @@
+                index_name=self.index_name,
+                embedding_function=self.embedding,
+                opensearch_url=self.opensearch_url,
+                http_auth=(self.username, self.password),
+                use_ssl=self.use_ssl,
+                verify_certs=self.verify_certs,
+                ssl_assert_hostname=False,
+                ssl_show_warn=False,
+            )
+        except Exception as e:
+            logger.error(f"Failed to create OpenSearchVectorSearch instance: {str(e)}")
+            raise RuntimeError(f"Check your connection and credentials: {str(e)}") from e
+
+        # If there are documents to ingest, add them to the vector store
+        if self.ingest_data:
+            self._add_documents_to_vector_store(opensearch)
+
+        return opensearch
+
+    def _add_documents_to_vector_store(self, vector_store: "OpenSearchVectorSearch") -> None:
+        """
+        Adds documents to the Vector Store.
+        """
+        documents = []
+        for _input in self.ingest_data or []:
+            if isinstance(_input, Data):
+                documents.append(_input.to_lc_document())
+            else:
+                error_message = f"Expected Data object, got {type(_input)}"
+                logger.error(error_message)
+                raise ValueError(error_message)
+
+        if documents and self.embedding is not None:
+            logger.debug(f"Adding {len(documents)} documents to the Vector Store.")
+            try:
+                vector_store.add_documents(documents)
+            except Exception as e:
+                error_message = f"Error adding documents to Vector Store: {str(e)}"
+                logger.error(error_message)
+                logger.error(f"Traceback: {traceback.format_exc()}")
+                raise RuntimeError(f"An error occurred while adding documents to the Vector Store: {str(e)}") from e
+        else:
+            logger.debug("No documents to add to the Vector Store.")
+
+    def search(self, query: Optional[str] = None) -> List[Dict[str, Any]]:
+        """
+        Search for similar documents in the vector store or retrieve all documents if no query is provided.
+        """
+        try:
+            vector_store = self.build_vector_store()
+
+            query = query or ""
+
+            if self.hybrid_search_query.strip():
+                # Use custom hybrid search query if provided
+                try:
+                    hybrid_query = json.loads(self.hybrid_search_query)
+
+                except json.JSONDecodeError as e:
+                    logger.error(f"Invalid hybrid search query JSON after replacement: {str(e)}")
+                    raise ValueError(f"Invalid hybrid search query JSON after replacement: {str(e)}") from e
+
+                results = vector_store.client.search(index=self.index_name, body=hybrid_query)
+
+                processed_results = []
+                for hit in results.get("hits", {}).get("hits", []):
```

Comment:
I'm not 100% familiar with this structure, am assuming you've tested and it works as expected

Reply:
Yes, I tested it and it worked fine.